static/js/recent_topics.js

```diff
@@ -536,6 +536,10 @@ export function show() {
     message_view_header.render_title_area();
 
     complete_rerender();
+    // once recent topic search input box and results are rendered , if input is not empty highlight contents of searchbox.
+    if ($("#recent_topics_search").val() !== "") {
+        $("#recent_topics_search").select();
+    }
```

Comment:
This should belong in `set_default_focus` since we don't want to steal focus away from the topic row if it was focused prior to this.

Reply:
@amanagr sorry for delayed reply and thanks for reviewing the code changes , tried adding check in set_deafault_focus , it looks like for each letter change text gets selected automatically which makes it difficult to type without text getting replaced and current usecase during topic change text is not getting highlighted.
![suggestion_fix](https://user-images.githubusercontent.com/10942163/114263636-70538300-9a04-11eb-9e96-b1237f5d836e.gif)

we want to select text only when there is a change in hash(any other topic to recent_topics).

Reply:
We don't want to set focus to search box every time we show recent topics and then set focus to `current_focus_element`. This is clearly evident from UI as well. So, we could set a variable in `show` that when `set_default_focus` is called next time,  we do `select` instead of `focus` and the set this variable to false.